### PR TITLE
support graphics images with v2.0 template

### PR DESCRIPTION
**Pull request by jhou5**

### CODE REVIEW

File: template/components/ospray-mpi.m4 [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+dnl CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+dnl OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+dnl OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+dnl
+include(begin.m4)
+
+DECLARE(`OSPRAY_MPI_VER',c42a885)
+
+ifelse(OS_NAME,ubuntu,`
+define(`OSPRAY_MPI_BUILD_DEPS',`ca-certificates g++ make cmake git libglfw3-dev libgl1-mesa-dri libxrandr-dev  libxinerama-dev libxcursor-dev libmpich-dev mpich openssh-server openssh-client')
```
Comment (xwu2git) [inline template/components/ospray-mpi.m4:36]: cmake needs to be conditional ifdef(`BUILD_CMAKE',,xxxx).

File: template/components/oiio.m4 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+dnl OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+dnl OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+dnl
+include(begin.m4)
+include(openexr.m4)
+
+DECLARE(`OIIO_VER',5daa9a1)
+
+ifelse(OS_NAME,ubuntu,`
+define(`OIIO_BUILD_DEPS',`ca-certificates g++ make cmake git')
```
Comment (xwu2git) [inline template/components/oiio.m4:37]: make cmake conditional on BUILD_CMAKE

File: template/components/oiio.m4 [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+')
+
+define(`BUILD_OIIO',`
+# build OpenImageIO
+
+ARG OIIO_REPO=https://github.com/OpenImageIO/oiio.git
+RUN git clone ${OIIO_REPO} && \
+    mkdir oiio/build && \
+    cd oiio/build && \
+    git checkout OIIO_VER && \
```
Comment (xwu2git) [inline template/components/oiio.m4:51]: git clone and git checkout can be simplified with git clone -b VER --depth 1 ${repo}

File: template/components/embree.m4 [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+dnl CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+dnl OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+dnl OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+dnl
+include(begin.m4)
+
+DECLARE(`EMBREE_VER',df0b324)
+
+ifelse(OS_NAME,ubuntu,`
+define(`EMBREE_BUILD_DEPS',`ca-certificates wget g++ make cmake git libtbb-dev libgl1-mesa-dev')
```
Comment (xwu2git) [inline template/components/embree.m4:36]: make cmake conditional on BUILD_CMAKE

File: template/components/oiio.m4 (hunk 2 above)
Comment (jhou5) [reply template/components/oiio.m4]: Fixed added - commit ID: 1a21f8a

File: template/components/embree.m4 (hunk 4 above)
Comment (jhou5) [reply template/components/embree.m4]: Fixed added - commit ID: 1a21f8a

File: template/components/oiio.m4 (hunk 3 above)
Comment (jhou5) [reply template/components/oiio.m4:51]: @xwu2git -b doesn't work for a specific commit. but if move to a specic branch/tag, source compiling will fail due to dependency changed.

**Review by xwu2git — APPROVED** [review]

File: template/components/embree.m4 [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+dnl FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+dnl DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+dnl SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+dnl CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+dnl OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+dnl OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+dnl
+include(begin.m4)
+
+DECLARE(`EMBREE_VER',df0b324)
```
[the remark this answers is not in the document]
Comment (jhou5) [reply template/components/embree.m4:33]: Build was failed in latest stable version. Will look for a fix when have bandwidth

File: Xeon/centos-7/graphics/dev/Dockerfile [hunk 6]
```diff
@@ -1,141 +1,131 @@
 
-FROM centos:centos7 AS build
-WORKDIR /home
 
-# COMMON BUILD TOOLS
-RUN yum install -y -q bzip2 make autoconf libtool git wget ca-certificates pkg-config gcc gcc-c++ bison flex patch epel-release yum-devel libcurl-devel zlib-devel;
```
[the remark this answers is not in the document]
Comment (jhou5) [reply Xeon/centos-7/graphics/dev/Dockerfile:6]: Good catch. I missed to insert it. will fix it soon.

**Pull request merged**